src/lib/drupal/queryResources.ts

```diff
@@ -0,0 +1,15 @@
+import { PHASE_PRODUCTION_BUILD } from 'next/constants'
+import { drupalClient } from '@/lib/drupal/drupalClient'
+import { QueryParams } from 'next-drupal-query'
+
+export async function getMenu(name: string, params: QueryParams<null>) {
+  const menu = await drupalClient.getMenu(name, {
+    params: params().getQueryObject(),
+
+    // Cache resource during build, not dev.
+    withCache: process.env.NEXT_PHASE === PHASE_PRODUCTION_BUILD,
+    cacheKey: `menu:${name}`,
+  })
+
+  return menu
+}
```

Comment:
>  I'm not sure that abstraction is necessary, though, because I now realize that there are really only two or three that would exist

This is correct, we really just need them for getMenu and getResource, if we want to abstract it. we can add cachekeys individually to the calls themselves, if we don't want the abstraction.

I do think nextjs's app router does allow caching at a slightly higher level than our fetches wrapped into next-drupal and the cache it can make use of, but this seemed like an appropriate spot for the time being